languages/python/default.nix

```
@@ -1,5 +1,21 @@
 { config, pkgs, lib, ... }:
-let cfg = config.programs.python;
+let
+  cfg = config.programs.python;
+  userWantsConfigPathEnv = cfg.configPath != null || cfg.config != null;
+  computedConfigPath = if cfg.configPath != null then
+    toString cfg.configPath
+  else
+    "${
+      if config.xdg.enable then
+        config.xdg.configHome
+      else
+        "${config.home.homeDirectory}/.config"
+    }/python/startup.py";
+  absoluteConfigPath =
+    if lib.hasPrefix config.home.homeDirectory computedConfigPath then
+      computedConfigPath
+    else
+      "${config.home.homeDirectory}/${computedConfigPath}";
```

```suggestion
      (if lib.hasPrefix "~" computedConfigPath then
        "${config.home.homeDirectory}/"
        + lib.removePrefix "~" computedConfigPath
      else
        "${config.home.homeDirectory}/${computedConfigPath}");
```

Just tested, looks good to me.

I'd just a test to remove the `~` prefix if present in `configPath`.

---

I don’t like this because it means the user can’t create a file called `~`.

---

(In this view the behavior of not duplicating a `${config.home.homeDirectory}` prefix is also questionable, but this can be easily rectified by not including the leading slash)